tests/api/v1/collectors/start.js

```diff
@@ -56,5 +56,80 @@ describe('tests/api/v1/collectors/start.js >', () => {
     .end(done);
   });
 
-  it('test invalid state changes');
+  it('Reject when the user token is invalid', (done) => {
+    api.post(path.replace('{key}', i))
+    .set('Authorization', 'iDontExist')
+    .send({})
+    .expect(constants.httpStatus.FORBIDDEN)
+    .expect((res) => {
+      expect(res.body.errors[0].description).to.equal('Invalid Token.');
+    })
+    .end(done);
+  });
+
+  // need token id to revoke it
+  it('Reject when the user token is revoked');
+
+  it('if the collector is not registered, throw an error.', (done) => {
+    Collector.findById(i)
+    .then((collector) => collector.update({ registered: false }))
+    .then(() => {
+      api.post(path.replace('{key}', i))
+      .set('Authorization', token)
+      .send({})
+      .expect(constants.httpStatus.FORBIDDEN)
+      .end(done);
+    });
+  });
+
+  it('if the collector is registered but status is PAUSED, ' +
+    'throw an error.', (done) => {
+    const _collector = JSON.parse(JSON.stringify(u.toCreate));
+    _collector.name = 'PausedCollector';
+
+    // if change from default status Stopped to Paused, will throw err
+    _collector.status = 'Paused';
+    Collector.create(_collector)
+    .then((c) => {
+      api.post(path.replace('{key}', c.id))
+      .set('Authorization', token)
+      .send({})
+      .expect(constants.httpStatus.FORBIDDEN)
+      .end(done);
+    });
+  });
+
+  it('if the collector is registered but status is RUNNING, ' +
+    'throw an error.', (done) => {
+    Collector.findById(i)
+    .then((collector) => collector.update({ status: 'Running' }))
+    .then(() => {
+      api.post(path.replace('{key}', i))
+      .set('Authorization', token)
+      .send({})
+      .expect(constants.httpStatus.FORBIDDEN)
+      .end(done);
+    });
+  });
+
+  it('if the collector is registered and status is STOPPED, set ' +
+    'status=RUNNING and return a collector token', (done) => {
+
+    // default status is STOPPED.
+    api.post(path.replace('{key}', i))
+    .set('Authorization', token)
+    .send({})
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        return done(err);
+      }
+
+      expect(res.body.status).to.equal('Running');
+      done();
+    });
+  });
+
+  it('if not found, create a new collector record with isRegistered=true and ' +
```

Comment:
Do we expect the users to POST with endpoint `v1/collectors/iDontExist/start` to make a collector called iDontExist? 
To be consistent with other endpoints, we would expect the response to be 404.
We can make a separate endpoint for `v1/collectors/start` to create a collector and start it.
What are your thoughts @iamigo @shriramshankar ?